core/chains/evm/types/models.go

```diff
@@ -227,6 +227,28 @@ func (h *Head) NextInt() *big.Int {
 	return new(big.Int).Add(h.ToInt(), big.NewInt(1))
 }
 
+// AsSlice returns a slice of heads up to length k
+// len(heads) may be less than k if the available chain is not long enough
+func (h *Head) AsSlice(k int) (heads []*Head) {
+	if k == 0 || h == nil {
+		return
+	}
+	heads = make([]*Head, 1)
+	heads[0] = h
+	for {
+		if len(heads) == k {
+			break
+		}
```

Comment:
Could use this form?
```suggestion
	for ; len(heads) < k; {
```